mirobo/vacuum.py

```diff
@@ -125,9 +133,12 @@ def send(self, command, parameters=None):
             _LOGGER.debug("%s:%s (ts: %s) << %s" % (self.ip, self.port,
                                                     m.header.value.ts,
                                                     m.data.value))
+            self._ts = m.header.value.ts
+            self._ts_server = int(time.mktime(datetime.datetime.utcnow().timetuple()))
             return m.data.value["result"]
         except Exception as ex:
             _LOGGER.error("got error when receiving: %s" % ex)
+            self.__enter__()
```

Comment:
What's the purpose of this? I don't think an exception handler should be doing this kind of logic.